File: pypiet.cpp
import <vector.h>


vector<int> d;

void psh(int x) {
    d.push_back(x);
}
int pop(int &x) {
    if (d.size()) {
        x = d.pop_back();
        return 1;
    }
    return 0;
}
int pop(int &x, int &y) {
    if (d.size() > 1) {
        x = d.pop_back();
        y = d.pop_back();
        return 1;
    }
    return 0;
}
void rll(int n, int k) {
    if (n > d.size() || n == 0 || k<1) {
        return;
    } else if ( n > 0 ){
        n = n%k;
        vector<int> z(d.begin(), d.begin+k);
        fill... d->d
        fill... z->d
    } else {
        n = n%k;
        vector<int> z(...);
        fill... d->d
        fill... z->d
    }
}





int main() {
    int t=0,i=0,a,b;
    char A;

    goto c_0_0_NOP;
%%%DEFN%%%
}

File: pypiet.py
stack = []

def psh(*X):
    for x in X:
        stack.append(x)

def pop():
    if stack:
        return stack.pop()
def pop2():
    if len(stack) > 1:
        return stack.pop(), stack.pop()
    else:
        return None, None

def rll(x, y):
    x %= y
    if not (y <= 0 or x == 0):
        z = -abs(x) + y * (x < 0)
        stack[-y:] = stack[z:] + stack[-y:z]


if __name__ == "__main__":
%%%DEFN%%%
    status = c_0_0_0_0
    while status is not None:
        status = status()
    
